From 8af84a16d8a70abe5b6379ecc8a37c0c333bcde4 Mon Sep 17 00:00:00 2001
From: Ivan Evlogiev <ivanevlogiev@codeaurora.org>
Date: Wed, 18 Sep 2013 10:00:55 +0300
Subject: [PATCH 040/176] QCamera2: Always get output crop for reprocessing
 stream

Get output crop even if thumbnail reprocessing is disabled

Change-Id: I5857668cdebdd60ad6af9dcb1ae11814a2c208d0
---
 QCamera2/HAL/QCamera2HWICallbacks.cpp | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/QCamera2/HAL/QCamera2HWICallbacks.cpp b/QCamera2/HAL/QCamera2HWICallbacks.cpp
index 0d16191..5e9a119 100644
--- a/QCamera2/HAL/QCamera2HWICallbacks.cpp
+++ b/QCamera2/HAL/QCamera2HWICallbacks.cpp
@@ -383,7 +383,7 @@ void QCamera2HardwareInterface::postproc_channel_cb_routine(mm_camera_super_buf_
     }
 
     if (pThmbStream == NULL) {
-        ALOGE("%s: Reprocessing stream not found", __func__);
+        ALOGE("%s: Reprocessing thumbnail stream not found", __func__);
     }
 
     for (int i = 0; i < recvd_frame->num_bufs; i++) {
@@ -400,7 +400,7 @@ void QCamera2HardwareInterface::postproc_channel_cb_routine(mm_camera_super_buf_
     }
 
     if (pSrcStream == NULL) {
-        ALOGE("%s: Snapshot stream doesn't exist, return here", __func__);
+        ALOGE("%s: Snapshot stream doesn't exist", __func__);
     }
 
     for (int i = 0; i < recvd_frame->num_bufs; i++) {
@@ -419,10 +419,10 @@ void QCamera2HardwareInterface::postproc_channel_cb_routine(mm_camera_super_buf_
     }
 
     if (pThmbSrcStream == NULL) {
-        ALOGE("%s: Thumbnail stream doesn't exist, return here", __func__);
+        ALOGE("%s: Thumbnail stream doesn't exist", __func__);
     }
 
-    if (pStream && pSrcStream && pThmbStream && pThmbSrcStream) {
+    if (pStream && pSrcStream) {
         cam_stream_parm_buffer_t param;
         memset(&param, 0, sizeof(cam_stream_parm_buffer_t));
         param.type = CAM_STREAM_PARAM_TYPE_GET_OUTPUT_CROP;
@@ -438,6 +438,12 @@ void QCamera2HardwareInterface::postproc_channel_cb_routine(mm_camera_super_buf_
                }
            }
         }
+    }
+
+    if (pThmbStream && pThmbSrcStream) {
+        cam_stream_parm_buffer_t param;
+        memset(&param, 0, sizeof(cam_stream_parm_buffer_t));
+        param.type = CAM_STREAM_PARAM_TYPE_GET_OUTPUT_CROP;
 
         rc = pThmbStream->getParameter(param);
         if (rc != NO_ERROR) {
-- 
1.8.3.1

